src/app/clusters/icd-management-server/icd-management-server.cpp

```diff
@@ -214,13 +238,15 @@ CHIP_ERROR CheckAdmin(CommandHandler * commandObj, const ConcreteCommandPath & c
 
 } // namespace
 
-/*
- * ICD Management Implementation
- */
-
 PersistentStorageDelegate * ICDManagementServer::mStorage           = nullptr;
 Crypto::SymmetricKeystore * ICDManagementServer::mSymmetricKeystore = nullptr;
-ICDConfigurationData * ICDManagementServer::mICDConfigurationData   = nullptr;
+
+void ICDManagementServer::InitCheckInServer(chip::PersistentStorageDelegate & storage,
```

Comment:
Do you mean that the `Init` signature should include all arguments for both scenarios? I created separate init for Check-in related functionalities, because otherwise the generic Init method requires including some headers like ICDMonitoringTable.h, which is not added as a dependency for SIT and also assigns the class members that are redundant.

Reply:
What were the pieces required the ICDMonitoringTable to be included in the generic init? Technically, we could include all the necessary headers without the `ICDMonitoringTable.h`, no?

Reply:
Ok, will address it.

Reply:
Done. I added few additional required includes and gn dependencies to make it work. Please take a look at it once again.